### ISSUE TITLE
invalid write when resetting device

### ISSUE THREAD

**Issue by tmakatos** [nutanix/libvfio-user#565]
Using SPDK and doing `echo 1 > /sys/bus/pci/devices/0000\:00\:03.0/reset` results in:
```
[2021-06-17 08:54:36.314321] vfio_user.c:1501:vfio_user_log: *DEBUG*: /var/tmp: W 7 0x70-0x72
[2021-06-17 08:54:44.710745] vfio_user.c:1509:vfio_user_log: *ERROR*: /var/tmp: failed to write 0x2-0x71: Invalid argument
```
This is a 2-byte write at `handle_px_write_2_bytes`, offset is 0x28.

**Comment by tmakatos** [comment]
Now we need to handle `struct pxlc2` (offset 0x30).

**Comment by tmakatos** [comment]
Fixed that, too. These two capabilities are re-written with their existing values, so we can probably ignore them for now.

**Comment by tmakatos** [comment]
Need to be using this: https://review.spdk.io/gerrit/c/spdk/spdk/+/8377/2.

Also, should fix this:
```
[2021-06-17 16:48:26.986482] vfio_user.c:1466:vfio_user_log: *ERROR*: /var/tmp: bad write command size 4
[2021-06-17 16:48:26.986506] vfio_user.c:1466:vfio_user_log: *ERROR*: /var/tmp: failed to write to PCI header: Invalid argument
[2021-06-17 16:48:26.986529] vfio_user.c:1466:vfio_user_log: *ERROR*: /var/tmp: failed to write 0x4-0x7: Invalid argument
[2021-06-17 16:48:26.986550] vfio_user.c:1466:vfio_user_log: *ERROR*: /var/tmp: msg0x20f: cmd 10 failed: Invalid argument
```

**Comment by tmakatos** [comment]
fixed by #569
